src/components/Select2.vue

```diff
@@ -72,63 +73,97 @@ export default {
     };
   },
   methods: {
-    toggleSelect: function(init) {
+    toggleSelect(init) {
       if (!init) {
         this.$refs.select.classList.toggle('active');
 
         Fliplet.Widget.autosize();
       }
     },
-    setOption: function(value, init) {
+    setOption(value, init) {
       this.setedOption = this.getSelectedOptionLabel(value);
-      this.$emit('onSelect', value);
+      this.$emit('update:selectedOption', value);
       this.toggleSelect(init);
     },
-    getSelectedOptionValue: function(value) {
+    getSelectedOptionValue(value) {
       if (!this.optionValueKey) {
         return value;
       }
 
       return value && value[this.optionValueKey];
     },
-    getSelectedOptionLabel: function(value) {
+    getSelectedOptionLabel(value) {
       if (!this.optionLabelKey) {
         return value;
       }
 
       return value && value[this.optionLabelKey];
     },
-    search: function() {
+    deepCopy(inObject) {
+      let outObject;
+      let value;
+      let key;
+
+      if (typeof inObject !== 'object' || inObject === null) {
+        return inObject; // Return the value if inObject is not an object
+      }
+
+      // Create an array or object to hold the values
+      outObject = Array.isArray(inObject) ? [] : {};
+
+      // eslint-disable-next-line guard-for-in
+      for (key in inObject) {
+        value = inObject[key];
+
+        // Recursively (deep) copy for nested objects, including arrays
+        outObject[key] = this.deepCopy(value);
+      }
+
+      return outObject;
+    },
+    search() {
+      // Using JSON.parse(JSON.stringify()) to make rip inner array bindings
+      const optionsCopy = this.deepCopy(this.options);
+
       if (this.customSearch) {
-        this.selectOptions = this.options.filter(option => {
-          return this.customSearch(this.searchValue, option);
-        });
+        if (this.selectWithGroups && this.selectOptions.length) {
```

Comment:
This section of logic looks like:

1. IF it's an array, DO X, which includes DO Y
1. ELSE DO Y

In this case, Y is written twice.

We often change this to:

1. IF it's not an array, turn it into an array
1. DO Y

Can we do that here?

Reply:
@tonytlwu actually `this.selectOptions` will be always an array, we check its length to find out was it an initial search to add options into the select.
So may I suggest that we better add an income parameter for the `search(init)` function and check if for false.
So our if would be like that ` if (this.selectWithGroups && !init)`
Would it be better?

Reply:
@YaroslavOvdii I don't understand what you mean but feel free to make changes as you fit. I'll review the result and and see whether it makes clear sense to me.

At the very least, if you what you said is true and `this.selectOptions` is always an array, then remove the unnecessary clauses.